tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/document/documents.controller.js

```diff
@@ -1162,7 +1140,9 @@
                             if (pinresponse.length > 0) {
                                 angular.forEach(pinresponse, function (pinobj) {
                                     angular.forEach(response, function (res) {
-                                        if (vm.changeDateFormat(pinobj.documentCreatedDate) == vm.changeDateFormat(res.documentCreatedDate)) {
+                                        var pinnedUrl = pinobj.documentParentUrl + "/" + pinobj.documentName
+                                        var searchDocumentUrl = res.documentParentUrl + "/" + res.documentName
+                                        if (pinnedUrl == searchDocumentUrl) {

```

Comment:
Instead of checking for the dates, going with the document names in the matter library and this will be always unique. The same logic has been used on the server side too